Mono.Addins.CecilReflector/Mono.Addins.CecilReflector/Reflector.cs

```diff
@@ -161,13 +161,13 @@ static void SetTypeNameAndAssemblyName (string basePropName, Type attype, object
 			if (prop == null)
 				throw new InvalidOperationException ("Property '" + typeNameProp + "' not found in type '" + attype + "'.");
 
-			prop.SetValue (ob, typeReference.FullName, null);
+			var assemblyName = typeReference.Resolve().Module.Assembly.FullName;
+			prop.SetValue (ob, typeReference.FullName + ", " + assemblyName, null);
 
-			prop = attype.GetProperty (basePropName + "AssemblyName", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
-			if (prop == null)
-				return;
+			prop = attype.GetProperty(basePropName + "FullName", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
```

Comment:
Do we really need to set this property? What is it for?

Reply:
This property is so we can query both assembly and typename, otherwise the type resolver will check the _current_ assembly for the type. Unless we pass the `AssemblyQualifiedName` here, the `AppDomain.AssemblyResolve` callbacks won't run.